src/components/Main.js

```diff
@@ -0,0 +1,25 @@
+import AddContacts from "./AddContacts";
+import IndividualContact from "./IndividualContact";
+import FullList from "./FullList";
+import { Routes, Route, } from 'react-router-dom';
+import React, { useState } from 'react';
+
+const Main = () => {
+  const  [contacts, setContacts] = useState([])
+
+  const mainState = (e) => {
```

Comment:
`mainState` is an interesting name for this. I would call it something like `updateSharedState` or something similar.